src/components/common/Modal.jsx

```diff
@@ -0,0 +1,41 @@
+import { createPortal } from 'react-dom';
+
+/* 사용방법
+1. 모달을 사용하고 싶은 곳에 import한다.
+2. <Modal>콘텐츠 내용</Modal> 을 입력한다.
+3. useModal() 훅을 사용한다.
+4. useModal()을 destructing해서 {isOpen, open, close}를 가져온다. {isOpen, open, close} = useModal();
+5. 모달 컴포넌트에 isOpen, close를 Props로 내려준다. isOpen={isOpen}, onClose(close)
+6. open은 모달을 열고 싶은 버튼에 onClick 이벤트로 넣어준다.
+7. 추가로 props로 title, button, extra를 보내줄 수 있다.
+*/
+
+const Modal = ({ title, button, children, extra, isOpen, onClose }) => {
+  if (!isOpen) return;

```

Comment:
이 문장은 `undefined`를 반환해서아무것도 렌더링하지 않긴 하지만, React 19부터는 렌더링이 함수 컴포넌트여야 한다는 원칙이 더 엄격해졌습니다. `return null;`을 명시적으로 쓰는 것이 안전하고 명확한 방식입니다!

```suggestion
  if (!isOpen) return null;
```